FreeRTOS/Test/CMock/queue/semaphore/counting_semaphore_utest.c

```diff
@@ -69,145 +69,6 @@ int suiteTearDown( int numFailures )
 
 
 
-/**
- * @brief Test xSemaphoreTake with an invalid counting semaphore
- * @details Verify that a call to xSemaphoreTake fails on a counting
- * semaphore created with uxMaxCount=0 and uxInitialCount=0
- * @coverage xQueueSemaphoreTake
- */
-void test_macro_xSemaphoreTake_CountingSemaphore_zero_zero_fail( void )
-{
-    /* Expect that xSemaphoreCreateCounting will assert because uxMaxCount=0 is invalid */
-    fakeAssertExpectFail();
-
-    SemaphoreHandle_t xSemaphore = xSemaphoreCreateCounting( 0, 0 );
-
-    fakeAssertGetNumAssertsAndClear();
-
-    /* validate returned semaphore handle */
-    TEST_ASSERT_NOT_EQUAL( NULL, xSemaphore );
-
-    TEST_ASSERT_EQUAL( QUEUE_T_SIZE, getLastMallocSize() );
-
-    /* Verify that an xSemaphoreTake fails */
-    TEST_ASSERT_EQUAL( pdFALSE, xSemaphoreTake( xSemaphore, 0 ) );
-
-    vSemaphoreDelete( xSemaphore );
-}
-
-/**
- * @brief Test xSemaphoreGive with an invalid counting semaphore
- * @details Verify that a call to xSemaphoreGive fails on a counting
- * semaphore created with uxMaxCount=0 and uxInitialCount=0
- * @coverage xQueueGenericSend
- */
-void test_macro_xSemaphoreGive_CountingSemaphore_zero_zero_fail( void )
-{
-    /* Expect that xSemaphoreCreateCounting will assert because uxMaxCount=0 is invalid */
-    fakeAssertExpectFail();
-
-    SemaphoreHandle_t xSemaphore = xSemaphoreCreateCounting( 0, 0 );
-
-    fakeAssertGetNumAssertsAndClear();
-
-    /* validate returned semaphore handle */
-    TEST_ASSERT_NOT_EQUAL( NULL, xSemaphore );
-
-    TEST_ASSERT_EQUAL( QUEUE_T_SIZE, getLastMallocSize() );
-
-    /* Verify that an xSemaphoreGive fails */
-    TEST_ASSERT_EQUAL( pdFALSE, xSemaphoreGive( xSemaphore ) );
-
-    vSemaphoreDelete( xSemaphore );
-}
-
-/**
- * @brief Test xSemaphoreGive with xSemaphoreCreateCounting( 1, 2 )
```

Comment:
One of these test cases is necessary for coverage of 
```
( uxInitialCount <= uxMaxCount ) )
```
in xQueueCreateCountingSemaphore